src/components/ConnectorToolbar.tsx

```diff
@@ -0,0 +1,264 @@
+import {
+    Autocomplete,
+    AutocompleteRenderInputParams,
+    FilledInputProps,
+    TextField,
+    Toolbar,
+} from '@mui/material';
+import { slate } from 'context/Theme';
+import { ConnectorWithTagDetailQuery } from 'hooks/useConnectorWithTagDetail';
+import { debounce } from 'lodash';
+import {
+    ChangeEvent,
+    Dispatch,
+    SetStateAction,
+    SyntheticEvent,
+    useMemo,
+    useRef,
+} from 'react';
+import { useIntl } from 'react-intl';
+import { CONNECTOR_NAME } from 'services/supabase';
+import { ENTITY, SortDirection } from 'types';
+
+interface Props {
+    cardWidth: number;
+    setColumnToSort: Dispatch<
+        SetStateAction<keyof ConnectorWithTagDetailQuery>
+    >;
+    setProtocol: Dispatch<SetStateAction<string | null>>;
+    setSortDirection: Dispatch<SetStateAction<SortDirection>>;
+    setSearchQuery: Dispatch<SetStateAction<string | null>>;
+}
+
+const inputProps: Partial<FilledInputProps> = {
+    disableUnderline: true,
+    sx: {
+        borderRadius: 5,
+        backgroundColor: (theme) =>
+            theme.palette.mode === 'dark'
+                ? 'linear-gradient(160deg, rgba(172, 199, 220, 0.18) 2%, rgba(172, 199, 220, 0.12) 40%)'
+                : slate[50],
+    },
+};
+
+function ConnectorToolbar({
+    cardWidth,
+    setColumnToSort,
+    setProtocol,
+    setSortDirection,
+    setSearchQuery,
+}: Props) {
+    const intl = useIntl();
+    const isFiltering = useRef(false);
+
+    const paramOptions: {
+        field: keyof ConnectorWithTagDetailQuery;
+        message: string;
+    }[] = useMemo(
+        () => [
+            {
+                field: CONNECTOR_NAME,
+                message: intl.formatMessage({
+                    id: 'connectorTable.data.title',
+                }),
+            },
+            {
+                field: 'image_name',
+                message: intl.formatMessage({
+                    id: 'connectorTable.data.image_name',
+                }),
+            },
+        ],
+        [intl]
+    );
+
+    const protocolOptions: {
+        protocol: ENTITY | null;
+        message: string;
+    }[] = useMemo(
+        () => [
+            {
+                protocol: null,
+                message: intl.formatMessage({
+                    id: 'common.optionsAll',
+                }),
+            },
+            {
+                protocol: ENTITY.CAPTURE,
+                message: intl.formatMessage({
+                    id: 'terms.capture',
+                }),
+            },
+            {
+                protocol: ENTITY.MATERIALIZATION,
+                message: intl.formatMessage({
+                    id: 'terms.materialization',
+                }),
+            },
+        ],
+        [intl]
+    );
```

Comment:
Same as below

Reply:
Same as the reply for `sortDirectionOptions`.